src/silicone/database_crunchers/quantile_rolling_windows.py

```diff
@@ -164,46 +158,51 @@ def derive_relationship(
                 xs = np.array([xs])
                 ys = np.array([ys])
 
-            step = (max(xs) - min(xs)) / (nwindows + 1)
+            step = (max(xs) - min(xs)) / nwindows
             decay_length = step / 2 * decay_length_factor
 
             sort_order = np.argsort(ys)
             ys = ys[sort_order]
             xs = xs[sort_order]
             if max(xs) == min(xs):
                 # We must prevent singularity behaviour if all the points are at the same x value.
-                window_centers = np.array([xs[0]] * nwindows)
-                decay_length = 1
+                cumsum_weights = np.array([(1 + x) / len(ys) for x in range(len(ys))])
+
+                derived_relationships[
+                    db_time
+                ] = lambda unused_variable, ys=ys, cumsum_weights=cumsum_weights, quantile=quantile: min(
+                    ys[cumsum_weights >= quantile]
+                )
```

Comment:
Can we change the lines above to the below (which is easier to understand I think)

```python
                def same_x_val_workaround(_, ys=ys, cumsum_weights=cumsum_weights, quantile=quantile):
                    return min(ys[cumsum_weights >= quantile])

                derived_relationships[
                    db_time
                ] = same_x_val_workaround
```

Reply:
This is in one of the new commits